### ISSUE TITLE
Building with openssl=no stills links libssl

### ISSUE THREAD

**Issue by akien-mga** [okamstudio/godot#1916]
Is this the intended behaviour?

I built export templates for x11 with ```openssl=no```, but in the end I still get libssl and libcrypto linked:
```
g++ -o bin/godot.x11.opt.debug.64.minimal platform/x11/godot_x11.x11.opt.debug.64.minimal.o platform/x11/context_gl_x11.x11.opt.debug.64.minimal.o platform/x11/os_x11.x11.opt.debug.64.minimal.o platform/x11/key_mapping_x11.x11.opt.debug.64.minimal.o -L/usr/lib64 main/libmain.x11.opt.debug.64.minimal.a modules/libmodules.x11.opt.debug.64.minimal.a bin/tests/libtests.x11.opt.debug.64.minimal.a drivers/libdrivers.x11.opt.debug.64.minimal.a drivers/libdrivers0.x11.opt.debug.64.minimal.a drivers/libdrivers1.x11.opt.debug.64.minimal.a scene/libscene.x11.opt.debug.64.minimal.a servers/libservers.x11.opt.debug.64.minimal.a core/libcore.x11.opt.debug.64.minimal.a -lX11 -lXinerama -lXcursor -lssl -lcrypto -lfreetype -lpulse-simple -lpulse -lGL -lGLU -lpthread -lasound -lz
```

Here is the full build command:
```
scons -j3 target=release_debug platform=x11 bits=default tools=no gdscript=yes vorbis=no minizip=no squish=no theora=no use_theoraplayer_binary=no freetype=no speex=no xml=yes png=yes jpg=no webp=no dds=no pvr=no etc1=no builtin_zlib=yes openssl=no musepack=no disable_3d=yes disable_advanced_gui=yes extra_suffix=minimal pulseaudio=yes
```

**Comment by akien-mga** [comment]
BTW the indentation here makes it look like openssl is a dependency of musepack (at least in my text editor, since the musepack line uses tabs and the openssl one uses spaces): https://github.com/okamstudio/godot/blob/master/SConstruct#L283-L286
The SConstruct script should probably be fixed to use only tabs (I could do that in a PR if need be).

**Comment by akien-mga** [comment]
Fixed by 2de5e12 and e1e54d3 I believe.
